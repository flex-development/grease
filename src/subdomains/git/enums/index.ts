/**
 * @file Entry Point - Enums
 * @module grease/git/enums
 */

export { default as RepositoryProvider } from './repository-provider'
