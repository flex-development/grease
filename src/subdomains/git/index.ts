/**
 * @file Entry Point - git
 * @module grease/git
 */

export * from './enums'
export { default as GitModule } from './git.module'
export type * from './interfaces'
export * from './models'
export * from './options'
export * from './providers'
export type * from './types'
