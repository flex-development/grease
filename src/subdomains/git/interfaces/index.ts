/**
 * @file Entry Point - Interfaces
 * @module grease/git/interfaces
 */

export type { default as ICommit } from './commit.interface'
