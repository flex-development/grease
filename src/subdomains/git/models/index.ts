/**
 * @file Entry Point - Models
 * @module grease/git/models
 */

export { default as CommitGrammar } from './commit-grammar.model'
export { default as Commit } from './commit.model'
export { default as Repository } from './repository.model'
