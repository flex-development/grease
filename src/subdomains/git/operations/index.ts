/**
 * @file Entry Point - Operations
 * @module grease/git/operations
 */

export { default as TagOperationHandler } from './tag.handler'
export { default as TagOperation, type TagOperationDTO } from './tag.operation'
