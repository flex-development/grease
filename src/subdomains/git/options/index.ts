/**
 * @file Entry Point - Options
 * @module grease/git/options
 */

export { default as CommitGrammarOptions } from './commit-grammar.options'
export { default as GitCommitOptions } from './git-commit.options'
export { default as GitTagOptions } from './git-tag.options'
export { default as GitOptions } from './git.options'
