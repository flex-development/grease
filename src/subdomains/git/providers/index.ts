/**
 * @file Entry Point - Providers
 * @module grease/git/providers
 */

export { default as GitService } from './git.service'
