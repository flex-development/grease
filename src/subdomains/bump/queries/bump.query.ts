/**
 * @file Queries - BumpQuery
 * @module grease/bump/BumpQuery
 */

import { GitOptions } from '#src/git'
import { get, type Partial } from '@flex-development/tutils'
import { IsString } from 'class-validator'

/**
 * Recommended version bump query.
 *
 * @see {@linkcode GitOptions}
 *
 * @class
 * @extends {GitOptions}
 */
class BumpQuery extends GitOptions {
  /**
   * Revision range end.
   *
   * @default 'HEAD'
   *
   * @public
   * @instance
   * @member {string} to
   */
  @IsString()
  public to: string

  /**
   * Create a new bump query.
   *
   * @param {Partial<BumpQuery>} [params] - Query parameters
   */
  constructor(params?: Partial<BumpQuery>) {
    super(params)
    this.to = get(params, 'to', 'HEAD')
  }
}

export default BumpQuery
