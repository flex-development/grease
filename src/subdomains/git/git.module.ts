/**
 * @file GitModule
 * @module grease/git/GitModule
 */

import { LoggerService, ValidationService } from '#src/providers'
import { Global, Module } from '@nestjs/common'
import { GitService } from './providers'

/**
 * Global git operations module.
 *
 * @class
 */
@Global()
@Module({
  exports: [GitService],
  providers: [GitService, LoggerService, ValidationService]
})
class GitModule {}

export default GitModule
