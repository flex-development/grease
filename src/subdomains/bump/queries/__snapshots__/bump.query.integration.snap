// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`integration:bump/queries/BumpQuery > validation > should fail validation if schema is invalid 1`] = `
[
  ValidationError {
    "children": [],
    "constraints": {
      "isString": "to must be a string",
    },
    "property": "to",
    "value": /\\^HEAD/,
  },
]
`;
